Mathlib/Tactic/Simproc/Apply.lean

```diff
@@ -0,0 +1,28 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import Lean.Elab.Tactic.Basic
+import Lean.Meta.Tactic.Simp.Main
+
+/-!
+# Procedure to execute a custom Simproc
+
+The only function `Simproc.apply` defined in this file takes a `s : Simproc` and executes it on the
+goal.
+-/
+
+open Lean Elab Tactic
+
+/-- Execute the given `Simproc` on the goal. -/
+def Lean.Meta.Simp.Simproc.apply (s : Simproc) : TacticM Unit := do
+  liftMetaTactic1 fun e ↦ do
+    -- `Simproc` usually does not allow arguments and can only be referred to by name, so we
+    -- hijacked `Simp.mainCore` to accept any `Simproc` term (i.e. not just a name), so that
+    -- a `Simproc` can accept arguments (by constructing a term of type `something -> Simproc`).
```

Comment:
This wording is confusing; `Simproc` doens't allow arguments any more than `Nat` allows arguments. I think what yoy mean is something like "we cannot pass a `Simproc` object into simpTheorems, which only accepts names of global constants"?

Reply:
changed the wording